docs/plugins.md

```diff
@@ -85,64 +85,224 @@ the output of Protractor by returning a results object.
+/**
+ * This is called inside browser.get() directly after the page loads, and before
+ * angular has had time to bootstrap/synchronize. 
+ *
+ * @this {Object} bound to module.exports
+ *
+ * @throws {*} If this function throws an error, a failed assertion is added to
+ *     the test results.
+ *
+ * @return {q.Promise=} Can return a promise, in which case protractor will wait
+ *     for the promise to resolve before continuing.  If the promise is
+ *     rejected, a failed assertion is added to the test results.
+ */
+exports.onPageLoad = function() {};
+
+/**
+ * This is called inside browser.get() directly after angular is done
+ * bootstrapping/synchronizing.  If browser.ignoreSynchronization is true, this
+ * will not be called.
+ *
+ * @this {Object} bound to module.exports
+ *
+ * @throws {*} If this function throws an error, a failed assertion is added to
+ *     the test results.
+ *
+ * @return {q.Promise=} Can return a promise, in which case protractor will wait
+ *     for the promise to resolve before continuing.  If the promise is
+ *     rejected, a failed assertion is added to the test results.
  */
-exports.postTest = function(config, passed, testInfo) {};
+exports.onPageSync = function() {};
+
+/**
+ * Between every webdriver action, Protractor calls browser.waitForAngular() to
+ * make sure that Angular has no outstanding $http or $timeout calls.
+ * You can use waitForPromise() to have Protractor additionally wait for your
+ * custom promise to be resolved inside of browser.waitForAngular().
+ *
+ * @this {Object} bound to module.exports
+ *
+ * @throws {*} If this function throws an error, a failed assertion is added to
+ *     the test results.
+ *
+ * @return {q.Promise=} Can return a promise, in which case protractor will wait
+ *     for the promise to resolve before continuing.  If the promise is
+ *     rejected, a failed assertion is added to the test results, and protractor
+ *     will continue onto the next command.  If nothing is returned or something
+ *     other than a promise is returned, protractor will continue onto the next
+ *     command.
+ */
+exports.waitForPromise = function() {};
+
+/**
+ * Between every webdriver action, Protractor calls browser.waitForAngular() to
+ * make sure that Angular has no outstanding $http or $timeout calls.
+ * You can use waitForCondition() to have Protractor additionally wait for your
+ * custom condition to be truthy.
+ *
+ * @this {Object} bound to module.exports
+ *
+ * @throws {*} If this function throws an error, a failed assertion is added to
+ *     the test results.
+ *
+ * @return {q.Promise<boolean>|boolean} If truthy, Protractor will continue onto
+ *     the next command.  If falsy, webdriver will continuously re-run this
+ *     function until it is truthy.  If a rejected promise is returned, a failed
+ *     assertion is added to the test results, and protractor will continue onto
+ *     the next command.
+ */
+exports.waitForCondition = function() {};
```

Comment:
Can we merge `waitForPromise` and `waitForCondition` into one method? Like how `webdriver.wait` can deal with both?